src/field.rs

```diff
@@ -143,4 +179,39 @@ mod test {
         assert_eq!(Fp31(<Fp31 as Field>::PRIME - 1), Fp31(0) - Fp31(1));
         assert_eq!(Fp31(0), Fp31(0) * Fp31(1));
     }
+
+    #[test]
+    fn pow() {
```

Comment:
These are great corner cases to test. Consider adding a basic test, e.g. 2^3 = 8, 3^3 = 27

Reply:
And something like 6^2 = 5.